### PR TITLE
Include old RecognisableSessions in matching

### PR DESCRIPTION
**Pull request by JonathanAndrews**
- check through all the user's previous RecognisableSessions for a matching RecognisableSessions.

### CODE REVIEW

File: app/controllers/devise_recognisable/sessions_controller.rb [hunk 1]
```diff
@@ -10,12 +10,12 @@ def perform_recognition_check
...
-    previous_session = DeviseRecognisable::RecognisableSession
+    previous_sessions = DeviseRecognisable::RecognisableSession
       .where( recognisable: self.resource )
-      .order(created_at: :desc).first
-    return unless previous_session.present?
+      .order(created_at: :desc)
+    return if previous_sessions.none?
 
-    unless Recogniser.recognise?(request, previous_session)
+    unless Recogniser.recognise?(request, previous_sessions)
```
Comment (jalada) [inline app/controllers/devise_recognisable/sessions_controller.rb:18]: To save passing around `request` within `Recogniser`, consider doing something like:

```
Recogniser.new(request).recognise?(previous_sessions)
```

That way you can save it as an instance variable.

If you wanna get real fancy:

```
Recogniser.with(request: request).recognise?(previous_sessions)
```

And `self.with` would just call `new`. Depends on how far down them 'make Ruby read like English' rabbit hole you wanna go.

File: spec/dummy-app/spec/features/sign_in_spec.rb [hunk 2]
```diff
@@ -162,4 +162,39 @@
       end
     end
   end
+
+  context 'older Recognisable sessions are still valid' do
```
Comment (jalada) [inline spec/dummy-app/spec/features/sign_in_spec.rb:166]: What does 'valid' mean for this test? Is it signing in using the older details? Newer ones?

**Comment by JonathanAndrews** [comment]
@jalada Can this be merged?

File: spec/dummy-app/spec/features/sign_in_spec.rb [hunk 3]
```diff
@@ -162,4 +162,39 @@
...
+      fill_in 'Email', with: user.email
+      fill_in 'Password', with: user.password
+      click_button 'Log in'
+      click_link 'Log out'
+
+      # Create a recognisable session from a different device.
+      FactoryBot.create :recognisable_session, different_recognisable_session_values
+    end
+
+    it 'works and does not send an email' do
```
Comment (jalada) [inline spec/dummy-app/spec/features/sign_in_spec.rb:189]: I'm still not 100% sure what this is testing... You're signing in, it creates a session record, then you make a new one with all new details, but it still let's them sign in. Oh wait...is it just testing that the system doesn't only look at the most recent session?

**Comment by jalada** [comment]
@JonathanAndrews code LGTM! Haven't tried running it on my machine but I'm sure it's fine 👍🏻

File: spec/dummy-app/spec/features/sign_in_spec.rb (hunk 3 above)
Comment (JonathanAndrews) [reply spec/dummy-app/spec/features/sign_in_spec.rb]: Yes. Third time lucky?

**Pull request merged**